### PR TITLE
Enable certain dependencies on RHEL/Alma 9

### PR DESCRIPTION
**Pull request by awegrzyn**

### CODE REVIEW

File: datadistribution.sh [hunk 1]
```diff
@@ -13,7 +13,7 @@ requires:
   - protobuf
   - O2
   - fmt
-  - "ucx:(slc8)"
+  - "ucx:(slc8|slc9)"
```
Comment (ktf) [inline datadistribution.sh:16]:
```suggestion
  - "ucx:(slc.*)"
```

File: mesos.sh [hunk 2]
```diff
@@ -16,7 +16,7 @@ requires:
   # - curl
   # - OpenSSL
 build_requires:
-  - "autotools:(slc6|slc7|slc8)"
+  - "autotools:(slc6|slc7|slc8|slc9)"
```
Comment (ktf) [inline mesos.sh:19]:
```suggestion
  - "autotools:(slc.*)"
```

File: o2suite.sh [hunk 3]
```diff
@@ -12,7 +12,7 @@ requires:
   - QualityControl
   - "DataDistribution:(?!osx)"
   - "ALF:(?!osx)"
-  - "mesos:(slc8)"
+  - "mesos:(slc8|slc9)"
```
Comment (ktf) [inline o2suite.sh:15]:
```suggestion
  - "mesos:(slc.*)"
```

**Review by ktf — COMMENTED** [review]
Any reason not to use a wildcard? In the end we do not support slc6 / slc7 anymore.

**Comment by awegrzyn** [comment]
> Any reason not to use a wildcard? In the end we do not support slc6 / slc7 anymore.

Yes, of course, it makes more sense.

**Review by ktf — DISMISSED** [review]

**Comment by awegrzyn** [comment]
@TimoWilken It seems that `build/O2/alidist` check running at `alientest02`  is missing a system dependency which it: `rdma-core-devel`

**Comment by TimoWilken** [comment]
Hi @awegrzyn, that check runs on slc7, so changing `slc8` to `slc.*` in `datadistribution.sh` means it also tries to build ucx on slc7, which it didn't before. Maybe for that one, it would be worth saying `slc[89]_` instead?

**Comment by awegrzyn** [comment]
Ah ok, thanks for feedback, I thought all checks are already at `slc8`.

**Pull request merged**